storage/rocksdb/rdb_comparator.h

```diff
@@ -41,27 +41,10 @@ class Rdb_pk_comparator : public rocksdb::Comparator {
   Rdb_pk_comparator &operator=(const Rdb_pk_comparator &) = delete;
   Rdb_pk_comparator() = default;
 
-  static int bytewise_compare(const rocksdb::Slice &a,
-                              const rocksdb::Slice &b) {
-    const size_t a_size = a.size();
-    const size_t b_size = b.size();
-    const size_t len = (a_size < b_size) ? a_size : b_size;
-    int res;
-
-    if ((res = memcmp(a.data(), b.data(), len)))
-      return res;
-
-    /* Ok, res== 0 */
-    if (a_size != b_size) {
-      return a_size < b_size ? -1 : 1;
-    }
-    return HA_EXIT_SUCCESS;
-  }
-
   /* Override virtual methods of interest */
 
   int Compare(const rocksdb::Slice &a, const rocksdb::Slice &b) const override {
-    return bytewise_compare(a, b);
+    return a.compare(b);
```

Comment:
It would be better if this was rocksdb::BytewiseComparator()->Compare(a, b);

Reply:
rocksdb::BytewiseComparator()->Compare is a virtual function call. Compare is one of the biggest cpu consumer function and RocksDB is very unlikely to change it.

```
virtual int Compare(const Slice& a, const Slice& b) const override {
    return a.compare(b);
  }
```